contracts/StorageManager.sol

```diff
@@ -46,48 +50,63 @@ contract StorageManager {
     // offerRegistry stores the open or closed Offer for provider.
     mapping(address => Offer) public offerRegistry;
 
+    // maps the tokenAddresses which can be used with this contract. By convention, address(0) is the native token.
+    mapping(address => bool) public isWhitelistedToken;
+
     event TotalCapacitySet(address indexed provider, uint64 capacity);
-    event BillingPlanSet(address indexed provider, uint64 period, uint128 price);
+    event BillingPlanSet(address indexed provider, address token, uint64 period, uint128 price);
     event MessageEmitted(address indexed provider, bytes32[] message);
 
     event NewAgreement(
         bytes32 agreementReference,
-        bytes32[] dataReference,
         address indexed agreementCreator,
         address indexed provider,
         uint128 size,
         uint64 billingPeriod,
         uint128 billingPrice,
         uint256 availableFunds
     );
-    event AgreementFundsDeposited(bytes32 indexed agreementReference, uint256 amount);
-    event AgreementFundsWithdrawn(bytes32 indexed agreementReference, uint256 amount);
-    event AgreementFundsPayout(bytes32 indexed agreementReference, uint256 amount);
+    event AgreementFundsDeposited(bytes32 indexed agreementReference, uint256 amount, address indexed token);
+    event AgreementFundsWithdrawn(bytes32 indexed agreementReference, uint256 amount, address indexed token);
+    event AgreementFundsPayout(bytes32 indexed agreementReference, uint256 amount, address indexed token);
     event AgreementStopped(bytes32 indexed agreementReference);
 
+    constructor() Ownable() Pausable() public { }
```

Comment:
Never seen this usage, is it setting correctly the deployer address as owner and pauser?

Reply:
Anyway all this will change with Upgradability as constructor will be removed with that change. So doesnt matter too much.